stubs/PyYAML/yaml/representer.pyi

```diff
@@ -1,60 +1,67 @@
-    alias_key: Any
-    def __init__(self, default_style=..., default_flow_style=..., sort_keys: bool = ...) -> None: ...
-    def represent(self, data): ...
+    default_flow_style: bool
+    represented_objects: dict[int, Node]
+    object_keeper: list[Any]
+    alias_key: int | Any
+    def __init__(self, default_style: str | None = ..., default_flow_style: bool = ..., sort_keys: bool = ...) -> None: ...
+    def represent(self, data) -> None: ...
+    def represent_data(self, data) -> Node: ...
     if sys.version_info < (3, 0):
         def get_classobj_bases(self, cls): ...
-    def represent_data(self, data): ...
     @classmethod
-    def add_representer(cls, data_type, representer): ...
+    def add_representer(cls: Type[_R], data_type: Type[_T], representer: Callable[[_R, _T], Node]) -> None: ...
     @classmethod
-    def add_multi_representer(cls, data_type, representer): ...
-    def represent_scalar(self, tag, value, style=...): ...
-    def represent_sequence(self, tag, sequence, flow_style=...): ...
-    def represent_mapping(self, tag, mapping, flow_style=...): ...
-    def ignore_aliases(self, data): ...
+    def add_multi_representer(cls: Type[_R], data_type: Type[_T], representer: Callable[[_R, _T], Node]) -> None: ...
+    def represent_scalar(self, tag: str, value, style: str | None = ...) -> ScalarNode: ...
+    def represent_sequence(self, tag: str, sequence: Sequence[Any], flow_style: bool | None = ...) -> SequenceNode: ...
+    def represent_mapping(self, tag: str, mapping: Mapping[Any, Any], flow_style: bool | None = ...) -> MappingNode: ...
+    def ignore_aliases(self, data) -> bool: ...
 
 class SafeRepresenter(BaseRepresenter):
-    def ignore_aliases(self, data): ...
-    def represent_none(self, data): ...
-    def represent_str(self, data): ...
+    inf_value: ClassVar[float]
+    def ignore_aliases(self, data) -> bool: ...
+    def represent_none(self, data) -> ScalarNode: ...
+    def represent_str(self, data: str) -> ScalarNode: ...
     if sys.version_info < (3, 0):
         def represent_unicode(self, data): ...
         def represent_long(self, data): ...
-    def represent_bool(self, data): ...
-    def represent_int(self, data): ...
-    inf_value: Any
-    def represent_float(self, data): ...
-    def represent_list(self, data): ...
-    def represent_dict(self, data): ...
-    def represent_set(self, data): ...
-    def represent_date(self, data): ...
-    def represent_datetime(self, data): ...
-    def represent_yaml_object(self, tag, data, cls, flow_style=...): ...
-    def represent_undefined(self, data): ...
+    def represent_binary(self, data: bytes) -> ScalarNode: ...
+    def represent_bool(self, data: bool) -> ScalarNode: ...
+    def represent_int(self, data: int) -> ScalarNode: ...
+    def represent_float(self, data: float) -> ScalarNode: ...
+    def represent_list(self, data: Sequence[Any]) -> SequenceNode: ...
+    def represent_dict(self, data: Mapping[Any, Any]) -> MappingNode: ...
```

Comment:
this broke 

```
return dumper.represent_dict(OrderedDict().items())
```

https://github.com/ceph/ceph/pull/41854/files#diff-c108f80763abd5f5682bc119e92da0f08f2f0749018a4c16ae1b281c462feec5R1039

Reply:
@sebastian-philipp dang, you're right, this is too restrictive.

Reply:
Maybe:

```suggestion
    def represent_dict(self, data: SupportsItems[Any, Any] | Iterable[tuple[Any, Any]]) -> MappingNode: ...
```

Reply:
@sebastian-philipp Is your code example from an open-source project? If it is, we could add it to mypy_primer (although it's currently used only for stdlib).

Reply:
We could add ceph to mypy_primer so that we won't break it in the future (although it currently doesn't help much, because only stdlib gets checked with mypy_primer). But I couldn't figure out how to mypy it:

```
akuli@akuli-desktop:~/sourcestuff/ceph$ mypy src/
mypy: can't decode file 'src/ceph-volume/ceph_volume/devices/lvm/migrate.py': 'utf-8' codec can't decode byte 0x92 in position 10665: invalid start byte
```

Reply:
ceph-volume doesn't yet have mypy coverage. 

Ceph is a C++ project with tons of Python code spread  across this mono repo. Python projects that already have mypy coverage are:

* https://github.com/ceph/ceph/blob/091a32e1302722e40f11079d3623e1aa79815e3f/qa/tox.ini#L11-L22
* https://github.com/ceph/ceph/blob/091a32e1302722e40f11079d3623e1aa79815e3f/src/pybind/tox.ini#L9-L14
* https://github.com/ceph/ceph/blob/091a32e1302722e40f11079d3623e1aa79815e3f/src/cephadm/tox.ini#L45
* https://github.com/ceph/ceph/blob/091a32e1302722e40f11079d3623e1aa79815e3f/src/pybind/mgr/tox.ini#L52-L98
* https://github.com/ceph/ceph/blob/091a32e1302722e40f11079d3623e1aa79815e3f/src/python-common/tox.ini#L5-L11

But I'd like to avoid you having deep link mypy-primer into those places directly. Do you need an entry point somewhere in the root dir?

Reply:
Yes, currently you need to list the directories. For example, my Porcupine project has this line in mypy_primer: https://github.com/hauntsaninja/mypy_primer/blob/c7c08f423364b87fdc856766e28ffe1c2f326e24/mypy_primer.py#L1241

Reply:
Sorry, I don't see an easy way to add Ceph to mypy_primer. There is nothing you can do in the repository's root dir that would help you to run mypy. :shrug: . 

An idea would be to add a tox integration to mypy-primer. We could then make sure that we reliably use the `mypy` tox environment. @hauntsaninja maybe?